### PR TITLE
ObjC tests: ARTRealtimeChannelHistoryTest

### PR DESCRIPTION
**Pull request by ricardopereira**
Tests passing:
 - [x] testHistory
 - [x] testHistoryBothChannels
 - [x] testHistoryForward
 - [x] testHistoryForwardPagination
 - [x] testHistoryBackwardPagination
 - [x] testTimeBackwards
 - [x] testTimeForwards
 - [x] testHistoryFromAttach

### CODE REVIEW

**Comment by ricardopereira** [comment]
- `ARTPaginatedResult.executePaginatedRequest` doesn't have authentication.
 - Added testRealtimeV2 helper.

**Comment by ricardopereira** [comment]
ARTPaginatedResult:
 - The request must be executed by the REST instance because it needs authentication.

All tests passing!

File: ably-iosTests/ARTTestUtil.m [hunk 1]
```diff
@@ -229,16 +230,40 @@ + (void)publishEnterMessages:(NSString *)clientIdPrefix count:(int) count channe
...
-+(void) testRealtime:(ARTRealtimeConstructorCb)cb {
++ (void)testRealtime:(ARTRealtimeConstructorCb)cb {
     [ARTTestUtil setupApp:[ARTTestUtil clientOptions] cb:^(ARTClientOptions *options) {
-        ARTRealtime * realtime = [[ARTRealtime alloc] initWithOptions:options];
+        ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
         cb(realtime);
     }];
 }
 
++ (void)testRealtimeV2:(XCTestCase *)testCase withDebug:(BOOL)debug callback:(ARTRealtimeTestCallback)callback {
```
Comment (mattheworiordan) [inline ably-iosTests/ARTTestUtil.m]: We could really do with a comment explaining why we have `testRealtime` and `testRealtimeV2`

File: ably-iosTests/ARTTestUtil.h [hunk 2]
```diff
@@ -39,10 +40,21 @@ typedef NS_ENUM(NSUInteger, TestAlteration) {
...
+// FIXME: try to unify testRealtime, testRealtimeV2 and others that are private
 + (void)testRealtime:(ARTRealtimeConstructorCb)cb;
 
+/**
+ New RealtimeClient instance
+
+ Creates implicitly a XCTestExpectation.
+ The callback is called only if the RealtimeClient gets connected.
+ */
++ (void)testRealtimeV2:(XCTestCase *)testCase withDebug:(BOOL)debug callback:(ARTRealtimeTestCallback)callback;
```
Comment (ricardopereira) [inline ably-iosTests/ARTTestUtil.h]: @mattheworiordan Done commenting.

**Pull request closed without merging**